### ISSUE TITLE
Android native code crashing when FirebaseMessaging.getInstance().getToken() fails 

### ISSUE THREAD

**Issue by RMG0** [aws-amplify/amplify-js#9528]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

Push Notifications

### Amplify Categories

notifications

### Environment information

<details>

```
# Put output below this line
  System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
    Memory: 3.43 GB / 15.71 GB
  Binaries:
    Node: 16.13.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.3.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 97.0.4692.99
    Edge: Spartan (44.19041.1266.0), Chromium (97.0.1072.76)
    Internet Explorer: 11.0.19041.1202
  npmPackages:
    @aws-amplify/pushnotification: ^4.3.8 => 4.3.8
    @babel/core: ^7.16.7 => 7.16.7
    @babel/runtime: ^7.16.7 => 7.16.7
    @react-native-community/eslint-config: ^3.0.1 => 3.0.1
    @react-native-community/netinfo: ^7.1.7 => 7.1.7
    @react-native-community/push-notification-ios: ^1.10.1 => 1.10.1 (1.0.3)
    amazon-cognito-identity-js: ^5.2.3 => 5.2.3
    aws-amplify: ^4.3.11 => 4.3.11
    axios: ^0.24.0 => 0.24.0 (0.21.4)
    babel-plugin-transform-remove-console: ^6.9.4 => 6.9.4
    eslint: 8.6.0 => 8.6.0
    eventing-bus: ^2.0.1 => 2.0.1
    hermes-inspector-msggen:  1.0.0
    metro-react-native-babel-preset: ^0.66.2 => 0.66.2
    patch-package: ^6.4.7 => 6.4.7
    react: 17.0.2 => 17.0.2
    react-native: 0.66.4 => 0.66.4
    react-native-cli: ^2.0.1 => 2.0.1
    react-native-device-info: ^8.4.8 => 8.4.8
    react-native-fast-image: ^8.5.11 => 8.5.11
    react-native-fs: ^2.18.0 => 2.18.0
    react-native-gesture-handler: 2.0.0 => 2.0.0
    react-native-haptic-feedback: ^1.13.0 => 1.13.0
  npmGlobalPackages:
    @aws-amplify/cli: 7.6.2
    @bugsnag/react-native-cli: 7.9.0
    @bugsnag/source-maps: 2.0.0
    appcenter-cli: 2.7.2
    npm: 8.3.2
    nswag: 13.7.0
    react-devtools: 4.8.2
    react-native-cli: 2.0.1
    yarn: 1.22.10

```

</details>

### On some android devices, app will crash when we getToken function fails

In  `RNPushNotificationModule `when we do `getToken`, when we fail to get a token in `getToken `function, we are passing exception back to js by `onErrorCallback.invoke(exception);` This will crash the app as exception object can not be passed through js bridge here and will crash like below stack trace

```
 java.lang.RuntimeException · Cannot convert argument of type class java.lang.Exception
Arguments.java:187 com.facebook.react.bridge.Arguments.fromJavaArgs
CallbackImpl.java:31 com.facebook.react.bridge.CallbackImpl.invoke
RNPushNotificationModule.java:86 com.amazonaws.amplify.pushnotification.RNPushNotificationModule$1.onComplete
```

### Expected behavior

Ideally we should pass just exception message like below
`onErrorCallback.invoke(exception.getMessage());`

### Reproduction steps

1. Run the app
2. App will crash on start up (Assuming we are initializing push notification on root mount)

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### Additional information and screenshots

**One can reproduce this by this test code simulating exception passing to js**

```
@ReactMethod
    public void getToken(final Callback onSuccessCallback, final Callback onErrorCallback) {
        FirebaseMessaging.getInstance().getToken().addOnCompleteListener(new OnCompleteListener<String>() {
                @Override
                public void onComplete(@NonNull Task<String> task) {
                    Exception exception = new Exception("MyTestExceptionMessage");
                    Log.e(LOG_TAG, "Error getting token: " + exception);
                    onErrorCallback.invoke(exception);
                }
            });
    }
```

**Comment by Samaritan1011001** [comment]
Thank you for catching this bug @RMG0 and your suggested solution seems to be the way to go. I have created a PR for the same.

**Comment by RMG0** [comment]
Thank you for PR and appreciate quick action.

**Comment by calebpollman** [comment]
@Samaritan1011001 Can you link the PR you created addressing this issue to this ticket please?

**Comment by Samaritan1011001** [comment]
Yeah, the [PR](https://github.com/aws-amplify/amplify-js/pull/9641) is under review and should be merged shortly.

**Comment by RMG0** [comment]
I think is now fixed and available in `"@aws-amplify/pushnotification": "^4.3.14"` release?
I can close this if this is true
